### PR TITLE
Fix sprite offset for pixel ratio !== 1

### LINKED ISSUE #12682: Using an Icon style with color makes the style's offset wrong  when the screen pixel ratio != 1
**Describe the bug**

When I have a vector source which I want to style using an image sprite, then the sprite offset is wrong in case the following is true:
1. I provide an offset and size to my icon style
2. My monitor has a pixel ratio > 1
3. I give a custom color to my icon style

**To Reproduce**
I made a reproduction here:
https://codepen.io/sebakerckhof/pen/zYzrzNO

In case I visit with my screen ratio = 1, I see this:
image

In case I visit with my screen ratio = 2, I see this:
image

In case I remove the custom color and keep my screen ratio = 2, I see this:
image

**Expected behavior**

The offset should select the same region of the image sprite no matter the screen ratio or color provided.

Update: This seems to have been introduced in 6.4.0, by this PR: https://github.com/openlayers/openlayers/pull/11277/files

But I'm not sure what's the correct path to fixing this.

### PR DESCRIPTION
**Pull request by ahocevar**
When changing the color of a sprite image, the sprite is scaled to match the render pixel ratio. This needs to be taken into account for the sprite offset.

Fixes #12682.

### CODE REVIEW

File: test/rendering/webpack.config.js [hunk 1]
```diff
@@ -42,6 +42,11 @@ export default {
     ],
   },
   resolve: {
+    fallback: {
+      fs: false,
+      http: false,
+      https: false,
+    },
```
Comment (ahocevar) [inline test/rendering/webpack.config.js:49]: This change is not related to the fix. It simply brings the rendering test webpack config in line with the other webpack configs. This avoids errors when building the rendering tests.

**Review by tschaub — APPROVED** [review]
